### ISSUE TITLE
[jvm-packages] unwanted training console output

### ISSUE THREAD

**Issue by kolarto** [dmlc/xgboost#4252]
When i create a Booster using `XGBoost.train()` method, there is this output to console:

> [0]	test-merror:0.330769	train-merror:0.191715

Where the errors of course correspond to test and train errors of the corresponding Booster model.
The problem is that the output shouldn't be there as i have set both parameters verbosity and silent to not to output anything.

```
xgParams.put("verbosity", 0);
xgParams.put("silent", 1);
```

My guess would be some forgotten System.out.print or something like that as this output is unaffected by logging settings.

**Comment by hcho3** [comment]
@kolarto A few things to check:
1. Make sure that the `watches` argument is empty.
2. Set `disable_default_eval_metric` parameter to 0

**Comment by kolarto** [comment]
Empty watches actually did the trick.
As i train iteratively using `booster.update()` and `booster.evalSet()` and use `XGBoost.train()` to only create the `Booster` and do the first iteration, I am fine.
But if I wanted to create and also train the booster using `XGBoost.train()`, what data does the booster actually use to mini/maximize evaluation metric when the `watches` are empty -> no reference to test data?
I though the main purpose of the `watches` is to set test data.

**Comment by hcho3** [comment]
> what data does the booster actually use to mini/maximize evaluation metric when the watches are empty 

The `XGBoost.train()` method will optimize the evaluation metric on the training data, not test data or other evaluation data. So you are able to perform training even with empty `watches`.
